### PR TITLE
Add member number 

### PR DESCRIPTION
**Pull request by reiterl**
Part of https://github.com/OpenSlides/OpenSlides/issues/6759

needs the backend pr https://github.com/OpenSlides/openslides-backend/pull/2412
and the meta pr in au https://github.com/OpenSlides/openslides-meta/pull/99

### CODE REVIEW

**Comment by reiterl** [comment]
Add export of member number and update the accounts export example file.

File: client/src/app/site/pages/organization/pages/accounts/pages/account-list/components/account-list/account-list.component.html [hunk 1]
```diff
@@ -63,7 +63,8 @@ <h2>{{ 'Accounts' | translate }}</h2>
                 </div>
                 <div class="meta-info user-subtitle ellipsis-overflow">
                     {{ user.saml_id || user.username }}
-                    <ng-container *ngIf="user.email">· {{ user.email }}</ng-container>
+                    <ng-container *ngIf="user.email">· {{ user.email }}&nbsp;</ng-container>
```
Comment (bastianjoel) [inline client/src/app/site/pages/organization/pages/accounts/pages/account-list/components/account-list/account-list.component.html:66]:
```suggestion
                    <ng-container *ngIf="user.email">&middot; {{ user.email }}&nbsp;</ng-container>
```
Change this here too.

**Review by bastianjoel — APPROVED** [review]

File: client/src/app/site/pages/organization/pages/accounts/pages/account-list/components/account-list/account-list.component.html (hunk 1 above)
Comment (reiterl) [reply client/src/app/site/pages/organization/pages/accounts/pages/account-list/components/account-list/account-list.component.html]: done

File: client/src/app/site/modules/user-components/components/user-detail-view/user-detail-view.component.html [hunk 2]
```diff
@@ -159,6 +159,14 @@ <h2>{{ 'Personal information' | translate }}</h2>
             </mat-form-field>
         </div>
 
+        <div *ngIf="isAllowed('update')">
+            <mat-form-field class="width-16p force-min-width">
+                <mat-label>{{ 'Membership number' | translate }}</mat-label>
+                <input type="text" matInput formControlName="member_number" />
+                <mat-error>{{ 'Membership number must be unique' | translate }}</mat-error>
```
Comment (Elblinator) [inline client/src/app/site/modules/user-components/components/user-detail-view/user-detail-view.component.html:166]: This Error is never activated.
If possible fix the activation 
If not possible please remove the line

**Review by Elblinator — CHANGES_REQUESTED** [review]
Please add the member_number to the participant export and the example file from the participant import

in participant-import change the preview title from 'member_number' to 'Member number'

in the participant-list: add the member_number to the  'edit details for "name"'- dialog

File: client/src/app/site/modules/user-components/components/user-detail-view/user-detail-view.component.html (hunk 2 above)
Comment (reiterl) [reply client/src/app/site/modules/user-components/components/user-detail-view/user-detail-view.component.html]: I have removed this mat-error. The uniqueness check is done in the backend. See username eg.

**Review by Elblinator — APPROVED** [review]

**Comment by Elblinator** [comment]
My approve was premature, please wait

**Review by Elblinator — CHANGES_REQUESTED** [review]
Add the Member number to the account filter

File: client/src/app/site/pages/organization/pages/accounts/services/common/account-filter.service.ts [hunk 3]
```diff
@@ -164,6 +164,14 @@ export class AccountFilterService extends BaseFilterListService<ViewUser> {
                         label: this.translate.instant(`Same email`)
                     }
                 ]
+            },
+            {
+                property: `hasMemberNumber`,
+                label: this.translate.instant(`Member number`),
```
Comment (Elblinator) [inline client/src/app/site/pages/organization/pages/accounts/services/common/account-filter.service.ts:170]: Please remove `this.translate.instant`
adding this creates the following bug: https://github.com/OpenSlides/openslides-client/issues/3580

I already created an issue for the preexisting filter terms (#3684

**Review by Elblinator — CHANGES_REQUESTED** [review]
see comment

File: client/src/app/site/pages/organization/pages/accounts/services/common/account-filter.service.ts (hunk 3 above)
Comment (reiterl) [reply client/src/app/site/pages/organization/pages/accounts/services/common/account-filter.service.ts]: Done, removed the translate.instant calls from the member number filter.

**Comment by reiterl** [comment]
Waiting on the backend/meta prs to be merged.

**Review by Elblinator — APPROVED** [review]

**Comment by bastianjoel** [comment]
When creating accounts on orga level an error `This member_number is forbidden` is displayed.

**Comment by bastianjoel** [comment]
@Elblinator have you tested removing a member number from a user?

**Comment by Elblinator** [comment]
Removing a member number results in the same snackbar Error ```This member_number is forbidden```

**Comment by Elblinator** [comment]
waiting for https://github.com/OpenSlides/openslides-backend/issues/2447

**Comment by reiterl** [comment]
Needs https://github.com/OpenSlides/openslides-backend/issues/2450 to work.

**Review by bastianjoel — APPROVED** [review]

**Review by Elblinator — APPROVED** [review]

**Pull request merged**